### PR TITLE
Added test for package

### LINKED ISSUE #2: Add unittest for the package

### PR DESCRIPTION
**Pull request by aadityasinha-dotcom**
fixes #2 
Consider this as my hacktober fest submission and please add `hacktoberfest-accepted` tag also.

### CODE REVIEW

**Comment by Agent-Hellboy** [comment]
@aadityasinha-dotcom Thanks for the PR

Here are some of the pointers

1 change your commit message 
   commit message should be ->  Add test for the package 
   please follow this 
   https://cbea.ms/git-commit/

 2 use pytest to write unittest  

 3  Unittest are mostly written considering in mind the edge case 

 4 In python, dict keys can be hashable so try to find what is the data structure which is hashable and tries to write the test for that

File: tests/test.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from trace_dkey import trace
```
Comment (Agent-Hellboy) [inline tests/test.py]: use pytest, pytest has fixture, which will be handy to write unit test for these kind of package

File: tests/test.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+def test_trace():
```
Comment (Agent-Hellboy) [inline tests/test.py]: test more scenarios, one of the example could be dict having keys as a tuple

**Review by Agent-Hellboy — CHANGES_REQUESTED** [review]

**Comment by Agent-Hellboy** [comment]
And the most important point is you should focus on increasing the code coverage, sorry I should put these in description of the issue

**Comment by aadityasinha-dotcom** [comment]
@Agent-Hellboy thanks for the review will make those changes

File: tests/test.py (hunk 1 above)
Comment (Agent-Hellboy) [reply tests/test.py]: you haven't used pytest yet, look at this to get an idea of `how I have used pytest`
https://github.com/Agent-Hellboy/log_call/blob/main/tests/test_log_call.py

File: tests/test.py [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
 
-def test_trace():
-    l={'a':{'b':{'c':{'d':{'e':{'f':1}}}}}}
+def test_trace_1():
+    l = {'a':{'b':{'c':{'d':{'e':{'f':1}}}}}}
     assert l['a']['b']['c']['d']['e']['f']==1
+
+def test_trace_2():
+    l = {'a':{'b':{'c':{'d':{'e':{'f':('h','i')},'g':1}}}}}
+    assert l['a']['b']['c']['d']['e']['f']==('h','i')
```
Comment (Agent-Hellboy) [inline tests/test.py]: I told you to check for a tuple as a key, not a value, this is a lib to track keys inside a dict

**Comment by aadityasinha-dotcom** [comment]
@Agent-Hellboy can you review the changes

**Review by Agent-Hellboy — CHANGES_REQUESTED** [review]

**Comment by Agent-Hellboy** [comment]
> @Agent-Hellboy can you review the changes

you haven't used pytest

**Comment by aadityasinha-dotcom** [comment]
> > @Agent-Hellboy can you review the changes
> 
> you haven't used pytest

@Agent-Hellboy   Added pytest fixtures can u review that.

**Comment by aadityasinha-dotcom** [comment]
@Agent-Hellboy can you also add `hacktoberfest` tag in your repository please

**Comment by Agent-Hellboy** [comment]
>@Agent-Hellboy Added pytest fixtures can u review that?

I don't know what kind of hurry you are in, you haven't tested any of the APIs, please do ask questions if are not able to move further or are not able to understand anything.

If I merge this PR then our code coverage will be 0%

I am coauthoring this PR as the current implementation had made writing unit test difficult, I hope you will understand something from this.

**Comment by Agent-Hellboy** [comment]
please check, https://github.com/Agent-Hellboy/trace-dkey/pull/6

**Comment by aadityasinha-dotcom** [comment]
Sorry

**Comment by aadityasinha-dotcom** [comment]
@Agent-Hellboy if u have any task u can assign me I am ready to work on.
